app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'person'
+require_relative 'classroom'
+require_relative 'student'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts "\nNo books yet"
+    else
+      puts "\nBooks:"
+      @books.each_with_index { |book, index| puts "#{index}: #{book.title} by #{book.author}" }
+    end
+  end
+
+  def list_all_persons
+    if @persons.empty?
+      puts "\nNo people yet"
+    else
+      @persons.each_with_index do |person, index|
+        puts "#{index}: [#{person.class}] name: #{person.name} ID: #{person.id} Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_a_person
+    loop do
+      print 'Would you like to create a student (1) or a teacher (2)? [Please enter the corresponding number]:'
+      option = gets.chomp.strip.to_i
+
+      case option
+      when 1
+        create_a_student
+        break
+      when 2
+        create_a_teacher
+        break
+      else
+        puts 'Option unknown'
+      end
+    end
+  end
+
+  def create_a_teacher
+    print 'Name: '
+    age = gets.chomp.to_i
+    print 'Age: '
+    name = gets.chomp
+    print 'Specialisation: '
+    specialization = gets.chomp
+    puts 'Teacher created successfully '
+    @persons.push(Teacher.new(name, age, specialization, parent_permission: true))
+  end
```

Comment:
![image](https://github.com/Zven94/OOP-school-library/assets/78251927/bfc47d25-8306-4a76-a296-d4844feca70e)

When I created a `Teacher`, the printed result did not correspond with the data input.  Please fix that 👍

Reply:
Hi @abdulhamiid . Sorry about that, the bugs were added in order to make a video using the pry gem with a partner. I forgot to change it after that